apps/android/app/src/main/java/chat/simplex/app/model/SimpleXAPI.kt

```diff
@@ -264,8 +264,17 @@ open class ChatController(var ctrl: ChatCtrl?, val ntfManager: NtfManager, val a
   private fun startReceiver() {
     Log.d(TAG, "ChatController startReceiver")
     if (receiverStarted) return
-    thread(name="receiver") {
-      GlobalScope.launch { withContext(Dispatchers.IO) { recvMspLoop() } }
+    receiverStarted = true
+    CoroutineScope(Dispatchers.IO).launch {
+      while (true) {
+        val ctrl = ctrl
```

Comment:
why is it needed?

Reply:
if the idea to capture ctrl from scope to avoid any concurrency issues by the time of the call, I would name the variable `_ctrl` and also add the comment why we are doing it.

Reply:
> I would name the variable _ctrl

Of course you would since you use Haskell:) But it's a Kotlin. In Haskell you wouldn't have this recursing problem at all. So, no need to copy Haskell decisions here 1:1

Reply:
So let’s just name it differently then, it’s not about Haskell, it’s a general hygiene to avoid shadowing